pkgs/applications/networking/p2p/magnetico/default.nix

```diff
@@ -0,0 +1,33 @@
+{ lib, fetchFromGitHub, buildGoModule, go-bindata }:
+
+buildGoModule rec {
+  pname = "magnetico";
+  version = "0.8.1";
+
+  src = fetchFromGitHub {
+    owner  = "boramalper";
+    repo   = "magnetico";
+    rev    = "v${version}";
+    sha256 = "1f7y3z9ql079ix6ycihkmd3z3da3sfiqw2fap31pbvvjs65sg644";
+  };
+
+  modSha256 = "1h9fij8mxlxfw7kxix00n10fkhkvmf8529fxbk1n30cxc1bs2szf";
+
+  buildInputs = [ go-bindata ];
+  buildPhase = ''
+    make magneticow magneticod
+  '';
+
+  doCheck = true;
+  checkPhase = ''
+    make test
```

Comment:
This should be the default, so no need to set `checkPhase` at all

Reply:
There is a difference, but I'm not sure why:

with `make test`
```
running tests
go test ./...
ok      github.com/boramalper/magnetico/cmd/magneticod  0.004s
ok      github.com/boramalper/magnetico/cmd/magneticod/bittorrent/metadata      0.004s
?       github.com/boramalper/magnetico/cmd/magneticod/dht      [no test files]
ok      github.com/boramalper/magnetico/cmd/magneticod/dht/mainline     0.005s
ok      github.com/boramalper/magnetico/cmd/magneticow  0.006s
ok      github.com/boramalper/magnetico/pkg/persistence 0.005s
?       github.com/boramalper/magnetico/pkg/util        [no test files]
```

with default `checkPhase`:
```
running tests
/nix/store/356zngkmiyiwq6ncnvgs9s32ps8mfml0-stdenv-linux/setup: line 1321: getGoDirs: command not found
```

Reply:
It seems `getGoDirs` is removed by the custom `buildPhase`. The problem is the build fails with `buildFlags = [ "magneticow" "magneticod" ]`.

Reply:
Oh this is a go package right, then it's not using the default stdenv phases. Yeah ignore what I said for the `make` parts, it doesn't apply here.

Reply:
All right.